src/modules/new-request-form/fields/LookupField.tsx

```diff
@@ -0,0 +1,245 @@
+      const newSelectedOption = {
+        name: custom_object_record.name,
+        value: custom_object_record.id,
+      };
+      setSelectedOption(newSelectedOption);
+      setInputValue(custom_object_record.name);
+      onChange(custom_object_record.id);
+    },
+    [customObjectKey, onChange]
+  );
+
+  const handleChange = useCallback<NonNullable<IComboboxProps["onChange"]>>(
+    async ({ inputValue, selectionValue }) => {
+      if (selectionValue !== undefined) {
+        if (selectionValue == "") {
+          setSelectedOption(EMPTY_OPTION);
+          setInputValue(EMPTY_OPTION.name);
+          setOptions([]);
+          onChange(EMPTY_OPTION.value);
+        } else {
+          const selectedOption = options.find(
+            (option) => option.value === selectionValue
+          );
+          if (selectedOption) {
+            setSelectedOption(selectedOption);
+            setInputValue(selectedOption.name);
+            onChange(selectedOption.value);
+            setOptions([selectedOption]);
+          }
+        }
+        return;
+      }
+
+      if (inputValue !== undefined) {
+        setInputValue(inputValue);
+        const searchParams = new URLSearchParams();
+        searchParams.set("name", inputValue.toLocaleLowerCase());
+        searchParams.set("source", "zen:ticket");
+        searchParams.set("field_id", fieldId.toString());
+        searchParams.set("requester_id", userId.toString());
+        if (organizationId !== null)
+          searchParams.set("organization_id", organizationId);
+        setIsLoadingOptions(true);
+        try {
+          const response = await fetch(
+            `/api/v2/custom_objects/${customObjectKey}/records/autocomplete?${searchParams.toString()}`
+          );
+
+          const data = await response.json();
+          if (data !== undefined) {
+            setOptions(
+              data.custom_object_records.map(
+                ({ name, id }: { name: string; id: string }) => ({
+                  name,
+                  value: id,
+                })
+              )
+            );
+          } else {
+            setOptions([]);
+          }
+        } catch (error) {
+          return error;
+        } finally {
+          setIsLoadingOptions(false);
+        }
+      }
+      return;
+    },
+    [customObjectKey, userId, organizationId, fieldId, onChange, options]
+  );
+
+  const debounceHandleChange = useMemo(
+    () => debounce(handleChange, 300),
+    [handleChange]
+  );
+
+  useEffect(() => {
+    return () => debounceHandleChange.cancel();
+  }, [debounceHandleChange]);
+
+  useEffect(() => {
+    if (value) {
+      fetchSelectedOption(value as string);
+    }
+  }, []);
+
+  const onFocus = () => {
+    setInputValue("");
+    selectedOption && setOptions([selectedOption]);
+  };
+
+  const handleKeyDown = useCallback((event) => {
+    if (event.key === "Enter") {
+      event.preventDefault();
+    }
+  }, []);
```

Comment:
Why do we need this? I don't think it is a good idea to override the default key bindings

Reply:
Because, when we hit enter in combobox we are submitting the whole form 🤷‍♀️ Not sure why it's only in this combobox, I've checked other fields also.

Reply:
I think it is the normal behavior of form inputs. If you press enter on any field (like subject, custom text input, custom date, cc field, ...) it will submit the form as well. The only fields where it is not happening are the WYSIWYG because the user needs to insert new lines and the other dropdowns because they don't have an input inside (we pass `isEditable={false}` as a prop).